frontend/src/EditUserProfile.tsx

```diff
@@ -0,0 +1,338 @@
+                </ToastAction>
+              ),
+            });
+        }
+      } else {
+        // Fallback to the default ErrorComponent
+        return <ErrorComponent error={error} />;
+      }
+    }
+  },
+});
+
+function EditUserProfile() {
+  const userQueryResult = useQuery(userQueryOptions);
+
+  if (userQueryResult.isFetching) {
+    <span>Loading...</span>;
+  }
+
+  if (userQueryResult.isError) {
+    return (
+      <span>
+        Unexpected error: {JSON.stringify(userQueryResult.error.message)} Please
+        report this{" "}
+        <a href="https://github.com/crux-bphc/chronofactorem-rewrite/issues">
+          here
+        </a>
+      </span>
+    );
+  }
+
+  const [firstDegree, setFirstDegree] = useState<string>(
+    userQueryResult.data.degrees[0],
+  );
+  const [secondDegree, setSecondDegree] = useState<string | null>(
+    userQueryResult.data.degrees.length > 1
+      ? userQueryResult.data.degrees[1]
+      : null,
+  );
+
+  let batch;
+  if (userQueryResult.data.email !== undefined) {
+    batch = userQueryResult.data.email.match(
+      /^f\d{8}@hyderabad\.bits-pilani\.ac\.in$/,
+    )
+      ? userQueryResult.data.email.slice(1, 5)
+      : "0000";
+  } else {
+    batch = "0000";
+  }
+
+  const mutation = useMutation({
+    mutationFn: (body: { degrees: (string | null)[] }) => {
+      return axios.post("/api/user/edit", body, {
+        headers: { "Content-Type": "application/json" },
+      });
+    },
+    onSuccess: () => {
+      router.navigate({ to: "/" });
+    },
+    onError: (error) => {
+      if (error instanceof AxiosError && error.response) {
+        if (error.response.status === 401) {
+          router.navigate({ to: "/login" });
+        } else if (error.response.status === 400) {
+          alert(`Error: ${error.message}`);
+        } else if (error.response.status === 500) {
+          alert(`Server error: ${error.message}`);
+        } else {
+          alert(`Server error: ${error.message}`);
+        }
+      }
+    },
+  });
+
+  const handleSubmit = async () => {
+    if (firstDegree.includes("B") && secondDegree === null) {
+      alert("Select your second degree!");
+      return;
+    }
+
+    const degrees = firstDegree.includes("B")
+      ? secondDegree !== null
+        ? secondDegree !== "A0"
+          ? [firstDegree, secondDegree]
+          : [firstDegree]
+        : [firstDegree]
+      : [firstDegree];
+
+    mutation.mutate({ degrees });
+  };
+
+  return (
+    <>
+      <div className="flex bg-slate-950 h-screen w-full justify-start">
+        <div className="flex flex-col lg:pt-28 pt-20 lg:mx-60 mx-12">
+          <h1 className="scroll-m-20 text-3xl tracking-tight lg:text-6xl text-slate-50 lg:mb-12 mb-4 font-bold">
+            Edit User Profile
+          </h1>
+          <div className="rounded-full text-slate-50 bg-slate-500 lg:px-9 lg:py-6 lg:text-6xl px-7 py-4 text-4xl lg:h-28 lg:w-28 h-20 w-20 lg:mx-4 mx-2 lg:mt-4 lg:mb-8 mt-2 mt-4">
```

Comment:
I agree this is brain damage, and instead, you can use a flexbox, it makes life much easier:
```suggestion
          <div className="rounded-full text-slate-50 bg-slate-500 lg:text-6xl text-4xl lg:h-28 lg:w-28 h-20 w-20 flex justify-center items-center">
```

Reply:
Done in [18609f1](https://github.com/crux-bphc/chronofactorem-rewrite/pull/117/commits/18609f1f5d8a202c1be08e707cf7a730855610e1)